README.md

```diff
@@ -69,88 +85,107 @@ describe('compliance', () => {
 
 # API
 
-A valid (read: that follows the interface defined) transport, must implement the following API.
+A valid transport (one that follows the interface defined) must implement the following API:
 
 **Table of contents:**
 
 - type: `Transport`
   - `new Transport([options])`
-  - `transport.dial(multiaddr, [options, callback])`
+  - `<Promise> transport.dial(multiaddr, [options])`
   - `transport.createListener([options], handlerFunction)`
   - type: `transport.Listener`
     - event: 'listening'
     - event: 'close'
     - event: 'connection'
     - event: 'error'
-    - `listener.listen(multiaddr, [callback])`
-    - `listener.getAddrs(callback)`
-    - `listener.close([options])`
+    - `<Promise> listener.listen(multiaddr)`
```

Comment:
Perhaps this should be a follow up PR instead, but since we're doing a breaking change it might be worth making `listen` support an array of multiaddrs now https://github.com/libp2p/interface-transport/issues/41.

Reply:
Makes sense... should `listen` require an array, or should it also support single multiaddr?

Reply:
I think it makes sense for it to just support an array. In the majority of cases the switch is going to be handling the `.listen` call, so most users shouldn't notice the change.

Reply:
I agree that we can leverage this breaking change and change the `multiaddr` param to an array. Should we do this in a separate PR, but only release a new version of `interface-transport` once that is done?

Reply:
@dirkmc I'm holding on the release until we've added the listening array support. Are you working on that? If not I can start it, just avoiding the potential duplicate effort :)

Reply:
@jacobheun please go ahead - I have some other things going on at the moment so only able to  snatch a bit of time here and there to work